gennotes_server/management/commands/add_all_clinvar_variants.py

```diff
@@ -76,8 +166,123 @@ def handle(self, *args, **options):
+            i += 1
+
+            ref_acc = rcva.find('ClinVarAccession')
+            rcv, rcv_ver = ref_acc.get('Acc'), ref_acc.get('Version')
+            rel_key = (rcv, rcv_ver)
+
+            if rel_key not in rcv_map:
+                # We do not have a record of this RCV from VCF, skip parsing...
+                continue
+
+            if len(rcv_map[rel_key]) != 1:
+                # either no or too many variations for this RCV
+                continue
+
+            val_store = {"type": "clinvar-accession",
+                         "clinvar-accession:id": rcv,
+                         "clinvar-accession:version": rcv_ver}
+            val_store["clinvar-accession:significance"] = rcva.findtext(
+                'ClinicalSignificance/Description')
+            val_store["clinvar-accession:disease-name"] = rcva.findtext(
+                'TraitSet[@Type="Disease"]/Trait[@Type="Disease"]/'
+                'Name/ElementValue[@Type="Preferred"]')
+
+            val_store['num_submissions'] = str(len(
+                ele.findall('ClinVarAssertion')))
+            val_store['record_status'] = rcva.findtext('RecordStatus')
+
+            ch = rcva.find('MeasureSet/Measure/'
+                           'MeasureRelationship[@Type="variant in gene"]')
+            if ch is not None:
+                val_store['gene:name'] = ch.findtext('Name/ElementValue')
+                val_store['gene:symbol'] = ch.findtext('Symbol/ElementValue')
+
+            val_store['citations'] = ';'.join(
+                ['PMID%s' % c.text for c in rcva.findall(
+                    'MeasureSet/Measure/Citation/ID[@Source="PubMed"]')])
+
+            val_store['xml:hash'] = self._hash_xml_dict(val_store)
+
+            if rel_key not in rcv_hash_cache:
+                # We got a brand new record
+                rel = Relation(variant=list(rcv_map[rel_key])[0],
+                               tags=val_store)
+                rel.save()
+                reversion.set_user(user=clinvar_user)
+                reversion.set_comment(
+                    'Relation added based on presence in XML file: ' +
+                    cv_filename)
+
+                rcv_hash_cache[rel_key] = (rel.pk, val_store['xml:hash'])
+
+                # TODO: set HGVS tag in Variant
+
+                print 'Added new:', i, rcv, rcv_ver,\
+                    rcv_map[rel_key], val_store['xml:hash']
+                for k, v in sorted(val_store.iteritems()):
+                    print '\t %s := %s' % (k, v)
+            elif rcv_hash_cache[rel_key][1] != val_store['xml:hash']:
+                # XML parameters have changed, update required
+                print 'Need to update', rcv, rcv_ver
+                rel = Relation.objects.get(**{
+                    'tags__clinvar-accession:id': rcv,
+                    'tags__clinvar-accession:version': rcv_ver,
+                    "tags__type": "clinvar-accession"})
+                rel.tags.update(val_store)
```

Comment:
rel.tags.update(val_store) just updates key that are found in the val_store dict. So it will overwrite only the tags found in the xml and defined in the parsing code above. Any custom tags will remain intact.